### PR TITLE
Add pw_unit_test support to build configuration

### PR DESCRIPTION
**Pull request by mbknust**
This is the first of a series of PRs, aiming to replace nlunit-test with pw_unit_test (https://github.com/project-chip/connectedhomeip/pull/29342, but split into smaller chunks).

### Changes

-   support for `pw_unit_test` added to `./.gn`
-   the GN template in `./build/chip/chip_test_suite.gni` changed, so it invokes `pw_test`
-   minor changes in `build/chip/chip_test.gni` and `build/chip/chip_test_group.gni`, to keep target naming consistent
-   small adjustments in `**/BUILD.gn`

### CODE REVIEW

File: build/chip/chip_test_suite.gni [hunk 1]
```diff
@@ -99,7 +99,104 @@ template("chip_test_suite") {
...
     }
   }
+  if (chip_link_tests) {
+    tests = []
+
+    if (defined(invoker.test_sources)) {
+      foreach(_test, invoker.test_sources) {
+        _test_name = string_replace(_test, ".cpp", "")
+
+        pw_test(_test_name) {
```
Comment (bzbarsky-apple) [inline build/chip/chip_test_suite.gni:109]: I'm trusting that this part is correct; I don't know enough about pw_test to tell.

File: third_party/silabs/gecko_sdk
Comment (bzbarsky-apple) [inline third_party/silabs/gecko_sdk:1]: I assume this change is not intended?

File: third_party/silabs/matter_support
Comment (bzbarsky-apple) [inline third_party/silabs/matter_support:1]: Likewise: why this change?

**Review by bzbarsky-apple — COMMENTED** [review]
Thank you for breaking this up into smaller pieces!

File: .gn [hunk 2]
```diff
@@ -29,6 +29,14 @@ default_args = {
   pw_build_PIP_CONSTRAINTS = [ "//scripts/setup/constraints.txt" ]
   pw_build_PIP_REQUIREMENTS = [ "//scripts/setup/requirements.build.txt" ]
 
+  pw_build_LINK_DEPS = [
```
Comment (tehampson) [inline .gn:32]: Just wondering if this could be added as a dependency to pw_test instead of as a default

File: build/chip/chip_test_suite.gni (hunk 1 above)
Comment (mbknust) [reply build/chip/chip_test_suite.gni:109]: At the very least, it seems to be working in practice - I've verified that when a suite uses pw_unit_test, running `./scripts/tests/all_tests.sh` will indeed cause the unit tests to get executed. The `pw_test` gn template (defined in `third_party/pigweed/repo/pw_unit_test/test.gni`) appears to be doing more or less the same thing as our very own `chip_test`, so it should be correct, unless I'm missing something.

File: third_party/silabs/gecko_sdk
Comment (mbknust) [reply third_party/silabs/gecko_sdk:1]: Yes, thanks for spotting this.

File: .gn (hunk 2 above)
Comment (mbknust) [reply .gn:32]: That's a good idea, but I don't know how to do this. I tried a few "obvious" approaches and none of them worked. Do you have something specific in mind?

**Review by bzbarsky-apple — APPROVED** [review]

File: .gn (hunk 2 above)
Comment (andy31415) [reply .gn]: @mbknust could you add a TODO for this here and then we can merge? it seems like how to achieve this is not immediately obvious however we should try to eventually have the dependency moved.

**Review by tehampson — APPROVED** [review]
Approval assumes adding the TODO with issue tracker. Just allowing this to go through without the need for me to revisit

File: .gn (hunk 2 above)
Comment (mbknust) [reply .gn]: > I am okay with just creating a TODO comment that points to a github issue
Would you prefer I create the issue before this is merged (so it can be referenced in the comment), or afterwards? This feels a bit chicken-and-egg.

Comment (bzbarsky-apple) [reply .gn]: For future reference, @mbknust : just create the issue before the PR merges, put link to the issue in the code.
